### PR TITLE
Cover commands with unit tests

### PR DESCRIPTION
**Pull request by okozlovlux**
- covered next commands with UT:
    ChangeRegistrationRequest,
    CommandRequestImpl,
    CommandNotificationFromMobileImpl,
    ResetGlobalPropertiesRequest,
    CreateInteractionChoiceSetRequest

Related to [APPLINK-28077](https://adc.luxoft.com/jira/browse/APPLINK-28077)

### CODE REVIEW

**Comment by okozlovlux** [comment]
@LevchenkoS @Kozoriz @LuxoftAKutsan @OHerasym @dtrunov @pvvasilev @wolfylambova22 please review

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 1]
```diff
@@ -419,18 +456,65 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByUITrue_SUCCESS) {
...
+      (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms];
+  EXPECT_CALL(*app_, vr_synonyms())
+      .WillOnce(Return(static_cast<smart_objects::SmartObject*>(&vr_synonyms)));
+  EXPECT_CALL(mock_app_manager_, hmi_capabilities())
+      .WillRepeatedly(ReturnRef(mock_hmi_capabilities_));
+
+  (*vr_synonyms_message)[am::strings::msg_params]
+                        [am::strings::hmi_display_language][0] = 1;
+  (*vr_synonyms_message)[am::strings::msg_params]
+                        [am::strings::hmi_display_language][1] = 2;
```
Comment (dtrunov) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux Magic numbers

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 2]
```diff
@@ -475,6 +559,64 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByVrTrue_SUCCESS) {
   command_->Run();
 }
 
+TEST_F(ChangeRegistrationRequestTest,
+       Run_IsLanguageSupportedByTTSLanguagesEmpty_SUCCESS) {
+  CreateDefaultMessage(message_);
+  (*message_)[am::strings::msg_params][am::strings::language] = 1;
+
```
Comment (dtrunov) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux magic number, please check all PR.

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 3]
```diff
@@ -374,6 +375,42 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByUI_UNSUCCESS) {
   command_->Run();
 }
 
+TEST_F(ChangeRegistrationRequestTest,
+       Run_IsLanguageSupportedByUIEmptyUILanguages_UNSUCCESS) {
+  CreateDefaultMessage(message_);
+  const utils::custom_string::CustomString kName(kAppName2);
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux, local const variables should be named as usual variables

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 4]
```diff
@@ -374,6 +375,42 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByUI_UNSUCCESS) {
...
+  application_set_.insert(app_);
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(app_));
+  EXPECT_CALL(mock_app_manager_, applications())
+      .WillOnce(Return(applications_));
+  EXPECT_CALL(*app_, name()).WillOnce(ReturnRef(kName));
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux, are this expectation is really important?

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 5]
```diff
@@ -419,18 +456,65 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByUITrue_SUCCESS) {
...
+      .WillOnce(Return(applications_));
+  EXPECT_CALL(*app_, name()).WillOnce(ReturnRef(name));
+
+  MessageSharedPtr vr_synonyms_message(CreateMessage());
+  (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms][0] =
+      kVrSynonyms2;
+  smart_objects::SmartObject& vr_synonyms =
+      (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms];
+  EXPECT_CALL(*app_, vr_synonyms())
+      .WillOnce(Return(static_cast<smart_objects::SmartObject*>(&vr_synonyms)));
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux, `static_cast` here is redundant and incorrect(you cast `SharedPtr<SmartObject>*` to  `SmartObject*`), please, rather use `vr_synonyms.get()`
Also, check everywhere, I saw this kind of issue in other places too

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 6]
```diff
@@ -475,6 +559,64 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByVrTrue_SUCCESS) {
   command_->Run();
 }
 
+TEST_F(ChangeRegistrationRequestTest,
+       Run_IsLanguageSupportedByTTSLanguagesEmpty_SUCCESS) {
+  CreateDefaultMessage(message_);
+  (*message_)[am::strings::msg_params][am::strings::language] = 1;
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux, why is `language` is `1`? As far as I know, languages have their own enum. Please check everywhere

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 7]
```diff
@@ -374,6 +375,42 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByUI_UNSUCCESS) {
...
+  (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms][0] =
+      kVrSynonyms2;
+  smart_objects::SmartObject& vr_synonyms =
+      (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms];
+  EXPECT_CALL(*app_, vr_synonyms())
+      .WillOnce(Return(static_cast<smart_objects::SmartObject*>(&vr_synonyms)));
+  EXPECT_CALL(mock_app_manager_, hmi_capabilities())
+      .WillRepeatedly(ReturnRef(mock_hmi_capabilities_));
+  smart_objects::SmartObjectSPtr ui_languages = CreateMessage();
+  (*ui_languages) = 0;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux why assignment with breakets?

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 8]
```diff
@@ -419,18 +456,65 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByUITrue_SUCCESS) {
...
 
+TEST_F(ChangeRegistrationRequestTest,
+       Run_IsLanguageSupportedByVRLanguagesEmpty_UNSUCCESS) {
+  CreateDefaultMessage(message_);
+  const utils::custom_string::CustomString name(kAppName2);
+  application_set_.insert(app_);
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(app_));
+  EXPECT_CALL(mock_app_manager_, applications())
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux prefere on_calls fir such situations

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 9]
```diff
@@ -419,18 +456,65 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByUITrue_SUCCESS) {
...
+  (*vr_synonyms_message)[am::strings::msg_params]
+                        [am::strings::hmi_display_language][1] = 2;
+  smart_objects::SmartObject* ui_languages =
+      &((*vr_synonyms_message)[am::strings::msg_params]
+                              [am::strings::hmi_display_language]);
+  EXPECT_CALL(mock_hmi_capabilities_, ui_supported_languages())
+      .WillOnce(Return(ui_languages));
+
+  smart_objects::SmartObjectSPtr vr_languages = CreateMessage();
+  (*vr_languages) = 0;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux do we have better syntax for creating smart object with 0? This one is ugly.

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 10]
```diff
@@ -475,6 +559,64 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByVrTrue_SUCCESS) {
...
+      .WillOnce(Return(applications_));
+  EXPECT_CALL(*app_, name()).WillOnce(ReturnRef(name));
+
+  MessageSharedPtr vr_synonyms_message(CreateMessage());
+  (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms][0] =
+      kVrSynonyms2;
+  smart_objects::SmartObject& vr_synonyms =
+      (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms];
+  EXPECT_CALL(*app_, vr_synonyms())
+      .WillOnce(Return(static_cast<smart_objects::SmartObject*>(&vr_synonyms)));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux  Why to you need static cast?

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 11]
```diff
@@ -475,6 +559,64 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByVrTrue_SUCCESS) {
...
+  EXPECT_CALL(mock_app_manager_, hmi_capabilities())
+      .WillRepeatedly(ReturnRef(mock_hmi_capabilities_));
+
+  (*vr_synonyms_message)[am::strings::msg_params]
+                        [am::strings::hmi_display_language][0] = 1;
+  (*vr_synonyms_message)[am::strings::msg_params]
+                        [am::strings::hmi_display_language][1] = 2;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux what is 1 what is 2? Magic numbers

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 12]
```diff
@@ -548,6 +690,142 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsNicknameAllowedFalse_UNSUCCESS) {
...
+       Run_IsNicknameAllowedInitResultFalse_UNSUCCESS) {
+  CreateDefaultMessage(message_);
+  (*message_)[am::strings::msg_params][am::strings::language] = 1;
+
+  const utils::custom_string::CustomString name(kAppName2);
+  application_set_.insert(app_);
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillRepeatedly(Return(app_));
+  EXPECT_CALL(mock_app_manager_, applications())
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux avoid expect call for such needs.

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 13]
```diff
@@ -263,35 +331,52 @@ TEST_F(ResetGlobalPropertiesRequestTest,
...
-          smart_objects::SmartType_Map);
-  EXPECT_CALL((*am::MockMessageHelper::message_helper_mock()),
-              CreateAppVrHelp(_)).WillOnce(Return(vr_help));
+  EXPECT_CALL(*app_, set_timeout_prompt(_));
+
+  smart_objects::SmartObjectSPtr prompt =
+      utils::MakeShared<smart_objects::SmartObject>();
+  (*prompt) = "prompt";
+  smart_objects::SmartObject* ptr = prompt.get();
+  EXPECT_CALL(*app_, timeout_prompt()).WillOnce(Return(ptr));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @okozlovlux code looks same as upper but variable name and value are prompt except timeout prompt.

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 14]
```diff
@@ -374,6 +375,42 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByUI_UNSUCCESS) {
...
 }
 
+TEST_F(ChangeRegistrationRequestTest,
+       Run_IsLanguageSupportedByUIEmptyUILanguages_UNSUCCESS) {
+  CreateDefaultMessage(message_);
+  const utils::custom_string::CustomString kName(kAppName2);
+  application_set_.insert(app_);
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(app_));
```
Comment (pvvasilev) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux Please rename `app_` to `mock_app_` for clarity.

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 1 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @dtrunov , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 2 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @dtrunov , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 3 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LevchenkoS , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 4 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LevchenkoS , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 6 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LevchenkoS , fixed 66da6e6

File: src/components/application_manager/test/commands/command_notification_from_mobile_impl_test.cc [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+  EXPECT_TRUE(command->CleanUp());
+  EXPECT_NO_THROW(command->Run());
+}
+
+TEST_F(CommandNotificationFromMobileImplTest, SendNotification_SUCCESS) {
+  MessageSharedPtr msg = CreateMessage();
+  CommandNotificationFromMobileImplPtr command =
+      CreateCommand<CommandNotificationFromMobileImpl>(msg);
+
+  EXPECT_CALL(*mock_message_helper_, PrintSmartObject(_))
```
Comment (okozlovlux) [inline src/components/application_manager/test/commands/command_notification_from_mobile_impl_test.cc]: @LuxoftAKutsan  , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 7 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LuxoftAKutsan  , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 8 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LuxoftAKutsan  , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 9 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LuxoftAKutsan  , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 10 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LuxoftAKutsan  , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 11 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LuxoftAKutsan  , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 12 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LuxoftAKutsan  , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 14 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LuxoftAKutsan  , fixed 66da6e6

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc (hunk 13 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @LuxoftAKutsan  , fixed 66da6e6

File: src/components/application_manager/test/commands/command_notification_from_mobile_impl_test.cc [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+ public:
+  CommandNotificationFromMobileImplTest()
+      : mock_message_helper_(am::MockMessageHelper::message_helper_mock()) {}
+
+  void SetUp() OVERRIDE {
+    Mock::VerifyAndClearExpectations(mock_message_helper_);
+  }
+
+ protected:
+  am::MockMessageHelper* mock_message_helper_;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/command_notification_from_mobile_impl_test.cc]: @okozlovlux some of you use message helpter by pointer, some by reference. Please use uniq styel. I prefers reference.

File: src/components/application_manager/test/commands/command_request_impl_test.cc [hunk 17]
```diff
@@ -397,6 +400,37 @@ TEST_F(CommandRequestImplTest,
...
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey;
+  (*msg)[strings::msg_params] = 0u;
+
+  CommandPtr command = CreateCommand<UCommandRequestImpl>(msg);
+
+  command->parameters_permissions().allowed_params.push_back(kAllowedParam);
+
+  SharedPtr<ApplicationSet> app_set;
+  MockAppPtr app(InitAppSetDataAccessor(app_set));
+  EXPECT_CALL(*app, app_id()).WillOnce(Return(kConnectionKey));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/command_request_impl_test.cc]: @okozlovlux please use ON_CALL's for such calls

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 18]
```diff
@@ -223,315 +259,304 @@ TEST_F(ChangeRegistrationRequestTest,
...
-
-  EXPECT_CALL(mock_app_manager_, application(_)).WillOnce(Return(app_));
-  EXPECT_CALL(mock_app_manager_, hmi_capabilities())
-      .WillOnce(ReturnRef(mock_hmi_capabilities_));
+  (*msg_)[am::strings::msg_params][am::strings::tts_name][0]
+         [am::strings::text] = "tts_name_1\n";
+  (*msg_)[am::strings::msg_params][am::strings::tts_name][1]
+         [am::strings::text] = "tts_name_2\n";
+
   EXPECT_CALL(mock_hmi_capabilities_, is_ui_cooperating()).Times(0);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux do any qctivities is rely not expected?

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 19]
```diff
@@ -223,315 +259,304 @@ TEST_F(ChangeRegistrationRequestTest,
...
-  smart_objects::SmartObject& name_synonyms =
-      (*name_synonyms_message)[am::strings::msg_params][am::strings::app_name];
-  EXPECT_CALL(*app_, vr_synonyms())
-      .WillOnce(
-          Return(static_cast<smart_objects::SmartObject*>(&name_synonyms)));
-  EXPECT_CALL(mock_app_manager_, hmi_capabilities())
-      .WillOnce(ReturnRef(mock_hmi_capabilities_));
+  smart_objects::SmartObject* name_synonyms =
+      &(*name_synonyms_message)[am::strings::msg_params][am::strings::app_name];
+  EXPECT_CALL(*mock_app_, vr_synonyms()).WillOnce(Return(name_synonyms));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux  ON_CALL

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc [hunk 20]
```diff
@@ -223,315 +259,304 @@ TEST_F(ChangeRegistrationRequestTest,
...
   smart_objects::SmartObject* vr_languages =
-      &((*vr_synonyms_message)[am::strings::msg_params]
-                              [am::hmi_response::languages]);
+      &(*vr_synonyms_message)[am::strings::msg_params]
+                             [am::hmi_response::languages];
   EXPECT_CALL(mock_hmi_capabilities_, vr_supported_languages())
       .WillOnce(Return(vr_languages));
 
+  smart_objects::SmartObjectSPtr tts_languages = CreateMessage();
+  *tts_languages = 0;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @okozlovlux do we have easyter way to create smart object with 0 value?

File: src/components/application_manager/test/commands/command_request_impl_test.cc (hunk 17 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/command_request_impl_test.cc]: @LuxoftAKutsan , fixed 	b5c884b

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 18 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LuxoftAKutsan , fixed 	b5c884b

File: src/components/application_manager/test/commands/mobile/change_registration_test.cc (hunk 19 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/change_registration_test.cc]: @LuxoftAKutsan , fixed 	b5c884b

**Pull request merged**